maven-extension/src/main/java/io/opentelemetry/maven/handler/MojoGoalExecutionHandler.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.maven.handler;
+
+import io.opentelemetry.api.trace.SpanBuilder;
+import org.apache.maven.execution.ExecutionEvent;
+
+public interface MojoGoalExecutionHandler extends Comparable<MojoGoalExecutionHandler> {
+
+  default void enrichSpan(SpanBuilder spanBuilder, ExecutionEvent executionEvent) {}
+
+  boolean supports(ExecutionEvent executionEvent);
+
+  default int ordinal() {
```

Comment:
Doesn't look like this value is overridden in any of the implementations. Were you intending for it to be set and used in a comparision?

Reply:
It's premature design 👼 
I used this concept in the Jenkins Pipeline Maven plugin ([here](https://github.com/jenkinsci/pipeline-maven-plugin/blob/pipeline-maven-parent-2.5.2/jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/MavenPublisher.java#L83)).

It relates to your question above about the matching handlers, do we invoke just the first matching or all matching.

My algorithm is inconsistent. Either we just invoke the first matching and then sort by ordinal to have predictability or we invoke all matching handlers and don't necessarily need to sort the handler even though predictability is very valuable.

Reply:
If we process multiple handlers, we'd likely end up with inconsistent, or possibly nonsensical, span attributes correct? It would be a mix of unique attributes from all handlers, with other values set based on the last one to set it.

Should probably only utilize one handler to enrich the span. So we either enforce distinctness on load, or provide an ordering and choose the first

Reply:
Change done, please verify new code.
I break at load time if multiple handlers overlap handling the same goal.